modules/services/preferences/src/main/java/au/com/shiftyjelly/pocketcasts/preferences/UserSetting.kt

```diff
@@ -61,28 +57,23 @@ abstract class UserSetting<T>(
     fun set(value: T, commit: Boolean = false, needsSync: Boolean = false) {
         persist(value, commit)
         _flow.value = value
-
-        if (needsSync) {
-            updateModifiedAtServerString()
-        }
+        val modifiedAt = if (needsSync) Instant.now().toString() else null
+        updateModifiedAtServerString(modifiedAt)
```

Comment:
The change here to set the modified at value to `null` if the setting does not need sync is the core change in this PR.